Assignment8/Tracking Application/Tracking Application/MainWindow.xaml.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Timers;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Data;
+using System.Windows.Documents;
+using System.Windows.Input;
+using System.Windows.Media;
+using System.Windows.Media.Imaging;
+using System.Windows.Navigation;
+using System.Windows.Shapes;
+using System.Windows.Threading;
+
+namespace TrackingApplication
+{
+    /// <summary>
+    /// Interaction logic for MainWindow.xaml
+    /// </summary>
+    public partial class MainWindow : Window
+    {
+        private const string StartTimer = "0:00:00";
+        public Timer Timer { get; set; }
+
+        public List<(string Timer, string Description)> Records { get; private set; }
+
+        public string CurrentDescription { get => DescriptionBox.Text; }
+
+        public MainWindow()
+        {
+            InitializeComponent();
+            Records = new();
+            Timer = new(1000);
+            Timer.Elapsed += Timer_Tick;
+        }
+
+        private void Timer_Tick(object sender, EventArgs e)
+        {
+            TimerAttributes.IncrementSecond();
+            Dispatcher.Invoke(() => TimerText.Text = TimerAttributes.GetString());
+        }
+
+        private void StartButton_Click(object sender, RoutedEventArgs e) => Timer.Start();
+
+        private void StopButton_Click(object sender, RoutedEventArgs e) => Timer.Stop();
+
+        private void SaveButton_Click(object sender, RoutedEventArgs e)
+        {
+            if(CurrentDescription.Length == 0 && TimerText.Text == StartTimer)
+                    return;
+            Timer.Stop();
+            Records.Add((TimerAttributes.GetString(), CurrentDescription));
+            TimerAttributes.ResetTimer();
+            DescriptionBox.Text = String.Empty;
+            TimerText.Text = StartTimer;
+            RecordsToApplication();
+        }
+
+        private void RecordsToApplication()
+        {
+            if(Records.Count > 0)
+            {
+                SavedTimerOne.Text = Records[^1].Timer;
+                SavedDescriptionOne.Text = Records[^1].Description;
+            }
+            else
+            {
+                SavedTimerOne.Text = String.Empty;
```

Comment:
Favor key works rather than class names when available.
```suggestion
                SavedTimerOne.Text = string.Empty;
```